Src/VimWpf/Implementation/MarkGlyph/MarkGlyphTagger.cs

```diff
@@ -0,0 +1,318 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Tagging;
+using Vim.Extensions;
+
+namespace Vim.UI.Wpf.Implementation.MarkGlyph
+{
+    internal sealed class MarkGlyphTagger : ITagger<MarkGlyphTag>, IDisposable

```

Comment:
Is there a reason you didn't use on of the tagger interface types here: in particular `BasicTaggerSource`? Those base types were written to handle all of the unwritten rules around taggers and handle most of them automatically. I've never written a glyph tagger though so it's possible those don't apply here.

Reply:
I discussed this in the "Notes" section of the PR description, which I will paste here:

> The original implementation of the mark glyph tagger attempted to use the `IBasicTaggerSource` infrastructure, but for reasons unknown, a glyph tagger apparently acts on the `VisualSnapshot` (elision buffer) vs. the `TextSnapshot` (normal text buffer). While this appeared to work, it caused large numbers of caught exceptions because the VsVim tagger infrastructure is not intended to work with the visual snapshot.

Reply:
I'm glad you asked because my conclusion was flawed as it is indeed possible to write a glyph tagger that operates on the normal text buffer and so I have reverted with success to the original implementation.